File: src/vanilla/DanModel/JavaInterface.cs
using System;

namespace AutoRest.Java.DanModel
{
    public class JavaInterface : JavaAbstractType
    {
        private readonly JavaFileContents contents;
        private bool addNewLine;

        public JavaInterface(JavaFileContents contents)
        {
            this.contents = contents;
        }

        private void AddExpectedNewLine()
        {
            if (addNewLine)
            {
                contents.Line();
                addNewLine = false;
            }
        }

        public void Method(string methodSignature)
        {
            AddExpectedNewLine();
            contents.Line(methodSignature + ";");

            addNewLine = true;
        }

        public void MultipleLineComment(Action<JavaMultipleLineComment> commentAction)
        {
            AddExpectedNewLine();
            contents.MultipleLineComment(commentAction);
        }

        public void SingleLineSlashSlashComment(string comment)
        {
            AddExpectedNewLine();
            contents.SingleLineSlashSlashComment(comment);
        }

        public void Annotation(params string[] annotations)
        {
            AddExpectedNewLine();
            contents.Annotation(annotations);
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/javamodel/JavaInterface.java
package com.azure.autorest.model.javamodel;

import java.util.function.Consumer;

public class JavaInterface implements JavaType
{
    private JavaFileContents contents;
    private boolean addNewLine;

    public JavaInterface(JavaFileContents contents)
    {
        this.contents = contents;
    }

    private void AddExpectedNewLine()
    {
        if (addNewLine)
        {
            contents.Line();
            addNewLine = false;
        }
    }

    public final void PublicMethod(String methodSignature)
    {
        PublicMethod(methodSignature, null);
    }

    public final void PublicMethod(String methodSignature, Consumer<JavaBlock> functionBlock)
    {
        AddExpectedNewLine();
        contents.Line(methodSignature + ";");

        addNewLine = true;
    }

    public final void JavadocComment(Consumer<JavaJavadocComment> commentAction)
    {
        AddExpectedNewLine();
        contents.JavadocComment(commentAction);
    }

    public final void LineComment(String comment)
    {
        AddExpectedNewLine();
        contents.LineComment(comment);
    }

    public final void Annotation(String... annotations)
    {
        AddExpectedNewLine();
        contents.Annotation(annotations);
    }
}